### PR TITLE
add fadeout as param in url

### PR DESCRIPTION
**Pull request by AvaYnE2**
:)

### CODE REVIEW

File: src/url.ts [hunk 1]
```diff
@@ -19,6 +19,16 @@ export function parseChannelFromURL() {
   return [channelID, channelLogin] as const;
 }
 
+const fadeoutValue = params.get("fadeout");
```
Comment (aidenwallis) [inline src/url.ts]: imo compose this into a single function, we can make this a lot more readable.


```ts

export const fadeout = resolveFadeout();

function resolveFadeout() {
   const raw = params.get("fadeout");
   if (raw === "off" || raw === "none") {
        return "none";
   }

   const value = parseInt(fadeout);
   if (isNaN(value) || value < 0) {
       return "15s";
   }

   return value;
}

Comment (aidenwallis) [reply src/url.ts]: Actually, if none just disables it. Make that return `null`

File: src/message.ts [hunk 2]
```diff
@@ -123,29 +119,29 @@ export class MessageElement extends LitElement {
       return html`<div class="emote-content">${renderFragment(firstEmote)}</div> `;
     }
 
+    const fadeOutAnimation = fadeout === "none" ? "none" : `fade-out 0.15s ease ${fadeout} forwards`;
```
Comment (aidenwallis) [inline src/message.ts]: thoughts on:

```ts
const fadeoutAnimation = fadeout ? `fade-out 0.15s ease ${fadeout} forwards` : "none";
```

**Review by aidenwallis — COMMENTED** [review]
Left some feedback 👍

**Comment by AvaYnE2** [comment]
Alright made the changes. Thanks for the feedback :)

File: src/url.ts [hunk 3]
```diff
@@ -19,6 +19,28 @@ export function parseChannelFromURL() {
...
+export const fadeout= resolveFadeout();
+function resolveFadeout() {
+  const raw = params.get("fadeout");
+  if (raw === "off" || raw === "none") {
+    return null;
+  }
+
+  if (!raw) {
+    return "15s";
+  }
```
Comment (aidenwallis) [inline src/url.ts:32]: do you need this? wouldn't `isNaN` return true for `""`

Comment (AvaYnE2) [reply src/url.ts:32]: Yeah technically it would work without it. But I have it because parseInt only takes string as param and raw it type string/null

**Review by aidenwallis — APPROVED** [review]

**Pull request merged**